packages/hop-node/src/watchers/BondWithdrawalWatcher.ts

```diff
@@ -347,6 +369,97 @@ class BondWithdrawalWatcher extends BaseWatcher {
+    await this.validateDbExistence(txParams)
+    await this.validateTransferSentIndex(txParams)
+    await this.validateUniqueness(txParams)
+    await this.validateLogsWithBackupRpc(txParams)
+  }
+
+  async validateDbExistence (txParams: SendBondWithdrawalTxParams): Promise<void> {
+    // Validate DB existence with calculated transferId
+    const calculatedDbTransfer = await this.getCalculatedDbTransfer(txParams)
+    if (calculatedDbTransfer.transferId !== txParams.transferId) {
+      throw new PreTransactionValidationError(`Calculated transferId (${calculatedDbTransfer?.transferId}) does not match transferId in db`)
+    }
+  }
+
+  async validateTransferSentIndex (txParams: SendBondWithdrawalTxParams): Promise<void> {
+    // Validate transferSentIndex is expected since it is not part of the transferId
+    const calculatedDbTransfer = await this.getCalculatedDbTransfer(txParams)
+    if (calculatedDbTransfer.transferSentIndex !== txParams.transferSentIndex) {
+      throw new PreTransactionValidationError(`transferSentIndex (${txParams.transferSentIndex}) does not match transferSentIndex in db (${calculatedDbTransfer.transferSentIndex})`)
+    }
+  }
+
+  async validateUniqueness (txParams: SendBondWithdrawalTxParams): Promise<void> {
+    // Validate uniqueness for redundant reorg protection. A transferNonce should be seen exactly one time in the DB per source chain
+    const txTransferNonce = txParams.transferNonce
+    const dbTransfers: Transfer[] = await this.db.transfers.getTransfersFromWeek()
+    const dbTransfersFromSource: Transfer[] = dbTransfers.filter(dbTransfer => dbTransfer.sourceChainId === this.bridge.chainId)
+    const transfersWithExpectedTransferNonce: Transfer[] = dbTransfersFromSource.filter(dbTransfer => dbTransfer.transferNonce === txTransferNonce)
+    if (transfersWithExpectedTransferNonce.length > 1) {
+      throw new PreTransactionValidationError(`transferNonce (${txTransferNonce}) exists in multiple transfers in db`)
+    }
+    if (transfersWithExpectedTransferNonce.length === 0) {
+      throw new PreTransactionValidationError(`transferNonce (${txTransferNonce}) does not exist in db`)
+    }
+  }
+
+  async validateLogsWithBackupRpc (txParams: SendBondWithdrawalTxParams): Promise<void> {
+    // Validate logs with backup RPC endpoint, if it exists
+    const calculatedDbTransfer = await this.getCalculatedDbTransfer(txParams)
+    const blockNumber = calculatedDbTransfer.transferSentBlockNumber
+
+    const redundantRpcUrls = getRedundantRpcUrls(this.chainSlug) || []
+    for (const redundantRpcUrl of redundantRpcUrls) {
+      const redundantProvider = new providers.JsonRpcProvider(redundantRpcUrl)
+
+      // TODO: Better way to do this
+      const l2Bridge = contracts.get(this.tokenSymbol, this.chainSlug)?.l2Bridge
+      const events = await l2Bridge.connect(redundantProvider).queryFilter(
+        l2Bridge.filters.TransferSent(),
```

Comment:
would it make sense to get event using `transferId` in filter?

```suggestion
        l2Bridge.filters.TransferSent(txParams.transferId),
```